host/tests/integration_tests/python/runtime/null_handling.rs

```diff
@@ -0,0 +1,141 @@
+use std::sync::Arc;
+
+use arrow::{
+    array::{Array, ArrayRef, Int64Array},
+    datatypes::{DataType, Field},
+};
+use datafusion_expr::{ColumnarValue, ScalarFunctionArgs, ScalarUDFImpl};
+
+use crate::integration_tests::{
+    python::test_utils::python_scalar_udf, test_utils::ColumnarValueExt,
+};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_no_optionals() {
+    const CODE: &str = "
+def add(x: int, y: int) -> int:
+    assert x is not None
```

Comment:
Feels kinda weird to me that we don't return an error if the udf errors. Do we just catch all errors and substitute them with `None`s? Is that configurable?